### PR TITLE
WT-2227 Resolve lint errors in join implementation

### PR DESCRIPTION
**Pull request by ddanderson**
@keithbostic, can you review and check with lint?  Thanks!

### CODE REVIEW

**Comment by keithbostic** [comment]
@ddanderson, one new one popped up:
```
< #... join_join(session, cjoin, idx, ref_cursor, flags, range, count, bloom_bit_
<           range, count, bloom_bit_count, bloom_hash_count));
< src/session/session_api.c  738  Info 734: Loss of precision (arg. no. 6) (32
<     bits to 8 bits)
```

File: src/session/session_api.c [hunk 1]
```diff
@@ -653,8 +653,9 @@ __session_join(WT_SESSION *wt_session, WT_CURSOR *join_cursor,
 	WT_INDEX *idx;
 	WT_SESSION_IMPL *session;
 	WT_TABLE *table;
-	uint64_t bloom_bit_count, bloom_hash_count, count;
-	uint32_t flags, range;
+	uint8_t flags;
+	uint32_t bloom_bit_count, bloom_hash_count, range;
+	uint64_t count;
```
Comment (sueloverso) [inline src/session/session_api.c]: @ddanderson swap `flags` and `count` lines so that they're in decreasing size order.

**Comment by ddanderson** [comment]
@keithbostic, @sueloverso, I think this should resolve both issues.

**Comment by ddanderson** [comment]
@keithbostic, thanks, I think this is ready.

**Comment by keithbostic** [comment]
Thanks Don, this looks great.

**Pull request merged**